tests/check_convert_util.c

```diff
@@ -64,9 +96,116 @@ START_TEST(test_convert_parse_header){
 }
 END_TEST
 
-START_TEST(test_convert_parse_tlvs){
-	/* TODO */
-	ck_assert(1);
+START_TEST(test_convert_parse_tlvs_generic){
+	int			ret;
+	struct convert_opts	opts;
+	uint8_t *		buff;
+	size_t			buff_len;
+	struct convert_tlv *	tlv;
+
+	buff	= (uint8_t *)sample_convert_connect(&buff_len);
+	tlv	= (struct convert_tlv *)buff;
+
+	ret = convert_parse_tlvs(buff, 0, &opts);
+	ck_assert_msg(ret == -1, "Should fail: 0-len buff");
+
+	ret = convert_parse_tlvs(buff, sizeof(struct convert_tlv) - 1, &opts);
+	ck_assert_msg(ret == -1,
+	              "Should fail: buff len shorter than TLV header");
+
+	ret = convert_parse_tlvs(buff, buff_len - 1, &opts);
```

Comment:
(this is more to test your own test functions but why not)

Reply:
This indeed assumes that sample_convert_connect() is correct :-) But assuming sample_convert_connect() is correct, this validates that convert_parse_tlvs() correctly detects a truncated packet. I could move the logic from sample_convert_error() to here, but the problem would remain the same in the end, the testing code has to be correct.